src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -956,15 +956,21 @@ public static bool IsDefaultExpressionContext(this SyntaxTree syntaxTree, int po
             var token = tokenOnLeftOfPosition;
             token = token.GetPreviousTokenIfTouchingWord(position);
 
-            // https://github.com/dotnet/roslyn/issues/39865: When the syntax rewrite is done, the parents here will need to change.
             switch (token.Kind())
             {
                 case SyntaxKind.LessThanToken:
                 case SyntaxKind.CommaToken:
-                    return token.Parent is FunctionPointerTypeSyntax;
+                    return token.Parent is FunctionPointerParameterListSyntax;
             }
 
-            return token.Parent is ParameterSyntax { Parent: FunctionPointerTypeSyntax _ };
+            return token switch
```

Comment:
You _might_ be able to merge this and the previous switch into a single pattern patch...

Reply:
I didn't want to duplicate the `parent is` check. When we can use C# 9 `or` patterns this can be cleaned up :)